### PR TITLE
Add Threading integration

### PR DESCRIPTION
**Pull request by geobeau**
Add a way to propagate tracer across thread. It was mentioned in issue #288.  Currently it doesn't actually trace the thread by themselves. It was done to not create unwanted span when we just need to trace other things across multiple child threads.

### CODE REVIEW

**Comment by geobeau** [comment]
CC: @bogdandrutu @liyanhui1228

File: opencensus/trace/ext/threading/trace.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+from opencensus.trace import execution_context
+
+log = logging.getLogger(__name__)
+
+MODULE_NAME = "threading"
+
+
+def trace_integration(tracer=None):
+    """Wrap the httplib to trace."""
```
[the remark this answers is not in the document]
Comment (geobeau) [reply opencensus/trace/ext/threading/trace.py]: Done

File: opencensus/trace/ext/threading/trace.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    log.info("Integrated module: {}".format(MODULE_NAME))
+    # Wrap the httplib request function
```
[the remark this answers is not in the document]
Comment (geobeau) [reply opencensus/trace/ext/threading/trace.py]: Done

File: opencensus/trace/ext/threading/trace.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    request_func = getattr(threading.Thread, "start")
+    setattr(threading.Thread, request_func.__name__,
+            wrap_threading_start(request_func))
+
+    request_func = getattr(threading.Thread, "run")
```
[the remark this answers is not in the document]
Comment (geobeau) [reply opencensus/trace/ext/threading/trace.py]: Done

File: opencensus/trace/ext/threading/__init__.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2017, OpenCensus Authors
```
[the remark this answers is not in the document]
Comment (geobeau) [reply opencensus/trace/ext/threading/__init__.py]: Done

File: README.rst [hunk 5]
```diff
@@ -428,6 +428,12 @@ You can enable Google Cloud client libraries integration by specifying ``'google
 
 .. _Cloud client libraries: https://github.com/GoogleCloudPlatform/google-cloud-python#google-cloud-python-client
 
+Threading
+~~~~~~~~~
+
+Census can propagate trace across threads when using the Threading package.
+
+You can enable Requests integration by specifying ``'threading'`` to ``trace_integrations``.
```
[the remark this answers is not in the document]
Comment (geobeau) [reply README.rst]: Done

**Comment by reyang** [comment]
Thank you @geobeau !

**Pull request merged**